packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -211,6 +215,17 @@ class ArtifactStore {
     return _engineRevision;
   }
 
+  static String _expectedEngineRevision;
+
+  static String get expectedEngineRevision {
+    if (_expectedEngineRevision == null) {
+      File revisionFile = new File(path.join(flutterRoot, 'packages', 'flutter', 'packages', 'sky_engine', 'REVISION'));
```

Comment:
I'm leaving a TODO for now, I agree that it's an issue, but out scope for this PR.

Filed https://github.com/flutter/flutter/issues/1509